### PR TITLE
Android Advanced Configuration Edits

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Provide a documentation with samples for the `RumResourceAttributesProvider`.

### CODE REVIEW

File: docs/configure_rum_android_sdk.md [hunk 1]
```diff
@@ -80,6 +80,53 @@ In addition to [tracking actions automatically][5], you can also track specific
 {{% /tab %}}
 {{< /tabs >}}
 
+### Enrich resources
+
+When [tracking resources automatically][6], you can provide a custom `RumResourceAttributesProvider` instance, allowing you to add custom attributes to each tracked network request. For example, if you want to track a network request's headers, you can create an implementation as follow, and pass it in the constructor of the `DatadogInterceptor`.
```
Comment (alai97) [inline docs/configure_rum_android_sdk.md:85]:
```suggestion
When [tracking resources automatically][6], provide a custom `RumResourceAttributesProvider` instance to add custom attributes to each tracked network request. For example, if you want to track a network request's headers, create an implementation as follows, and pass it in the constructor of the `DatadogInterceptor`.
```

**Review by alai97 — APPROVED** [review]
Looks good! Will there be a doc update to iOS Advanced Configuration? We can port over these changes to that page too.

File: docs/configure_rum_android_sdk.md [hunk 2]
```diff
@@ -80,9 +80,56 @@ In addition to [tracking actions automatically][5], you can also track specific
...
+            @NonNull Request request,
+            @Nullable Response response,
+            @Nullable Throwable throwable
+    ) {
+        Map<String, Object> result = new HashMap<>();
+        Headers headers = request.headers();
+
+        for (String key : headers.names()) {
+            String attrName = "headers." + key.toLowerCase(Locale.US);
+            result.put(attrName, headers.values(key));
```
Comment (0xnm) [inline docs/configure_rum_android_sdk.md:120]: java snippet doesn't exactly match kotlin snippet: in Java we add all the values for the header name, in kotlin we add only a single value.

Comment (xgouchet) [reply docs/configure_rum_android_sdk.md]: Fixed

**Review by 0xnm — APPROVED** [review]

**Pull request merged**